### DISCUSSION TITLE
JSON-RPC Configuration

### DISCUSSION THREAD

**Discussion by qlpqlp** [dogecoin/dogecoin#2615] [Help needed] [answered]
Hello to all Dev Shibs :)

Im starting a colections of LIB's and some tools for real case usage like Payments Bridges, Social Media Tools, Notifications, Nodes Upgrade notifications and "Much" more :)

Im trying to set an RCP connections to my nodes, and started reading the Bitcoin reference because there is still no extensive docs on Dogecoin GIT or dogecoin.com.

Im trying to set using Windows Dogecoin Core wallet with the starting command:
"C:\Program Files\Dogecoin\dogecoin-qt.exe" -conf=dogecoin.conf

My dogecoin.conf is like this exemple:

rpcuser=doge
rpcpassword=wow
addnode=192.168.1.1
rpcport=22555
server=1
daemon=1
paytxfee=0.01

But it its semans is not starting:

C:\Program Files\Dogecoin\daemon>dogecoin-cli getrpcinfo
error code: -32601
error message:
Method not found

I also have several linux nodes, that I still didnt tested, and maybe is only a windows limitation. Can anyone confirm the correct configurations and/or limitations?

Thanks in advance to all Dev's :)
@inevitable360

**Comment by patricklodder** [comment]
Hi!

Try `dogecoin-cli help` for an overview of all accepted methods and `dogecoin-cli help <command>` for details about the command.

The `getrpcinfo` method is not available until 1.21 for Dogecoin - what were you trying to find out?

**Reply by qlpqlp** [reply]
Hello @patricklodder ;

Yes, I have read the command list on dogecoin-cli :) 

I was trying to start the Dogecoin RPC server to be able to send external CURL commands.

I could code a server to relay to dogecoin-cli on windows but its an extra layer of security lol :)

On linux, im using my own Docker image, and I can easly send CURL commands to Docker and they relay directly to dogecoin-cli, so its easy to workaround, but on windows its a bit tricky :)

I'm using as reference this: https://en.bitcoin.it/wiki/API_reference_(JSON-RPC)#PHP

Thanks again :)

**Reply by patricklodder** [reply]
Does it work with `getinfo` though?

**Reply by qlpqlp** [reply]
yes i works the  getinfo command directly, but using this method:

curl --user doge --data-binary '{"jsonrpc": "1.0", "id": "curltest", "method": "getinfo", "params": []}' -H 'content-type: text/plain;' http://192.168.0.1:22555/

no, cannot resolve because the dogecoin rpc is not running or i missing something :)

I will try to make some more tests and also review the Dogecoin 1.14.4 code to try an alternative way, for now :)

Until 1.21 eheh :) hope to be able to compile and run it soon :)

**Comment by patricklodder** [comment]
> curl --user doge --data-binary '{"jsonrpc": "1.0", "id": "curltest", "method": "getinfo", "params": []}' -H 'content-type: text/plain;' 

```
http://192.168.0.1:22555/
```

You'll need to bind the RPC to that interface, by default it only does localhost. Add `rpcbind=<dogecoind addr>` and `rpcallowip=<curl host addr>` to your dogecoin.conf. Be careful to not expose a wallet though!!!

**Reply by qlpqlp** [reply]
Thanks one again @patricklodder for helping me :)
 
I will try that and post here the results on Thursday :) tomorow have to go to a DC in Lisbon lol

**Reply by qlpqlp** [reply]
Ok today I have tested and still no luck :(

`C:\Program Files\Dogecoin\daemon>dogecoin-cli -rest 1
error: couldn't connect to server: EOF reached (code 1)
(make sure server is running and you are connecting to the correct RPC port)`

my config

`rpcuser=doge
rpcpassword=wow
addnode=192.168.1.80
rpcport=22555
server=1
daemon=1
paytxfee=0.01
rpcbind=192.168.1.80
rpcallowip=192.168.1.80`

P.S- Firewall disable for testing

Will try later enabling the "Windows Subsystem for Linux" to give it a try , if not, will run VirtualBox on my Laptop or in last resort use one of my Servers on DC to develop/test on Linux :)

**Comment by patricklodder** [comment, marked as answer]
You probably need to spec `-rpcconnect=192.168.1.80` on your cli command. `-rest` doesn't do anything on cli.

**Reply by qlpqlp** [reply]
Much Thanks! Such coder! :)
![rcp_tested](https://user-images.githubusercontent.com/5968076/137503628-c8fb999a-1119-4cef-9256-ec970dece6b0.jpg)

Now it comunicates sucessfull with Dogecoin Nodes!

**Reply by anthonysidesauthor** [reply]
Thank old dipper
